src/Admin/Ide/Ide.php

```diff
@@ -0,0 +1,581 @@
+		$asset_file = include WPGRAPHQL_PLUGIN_DIR . 'build/ide.asset.php';
+		$render_asset_file = include WPGRAPHQL_PLUGIN_DIR . 'build/render.asset.php';
+		$graphql_asset_file = include WPGRAPHQL_PLUGIN_DIR . 'build/graphql.asset.php';
+
+		$app_context = $this->get_app_context();
+
+		wp_register_script(
+			'graphql',
+			plugins_url( 'build/graphql.js', WPGRAPHQL_PLUGIN_FILE ),
+			$graphql_asset_file['dependencies'],
+			$graphql_asset_file['version'],
+			false
+		);
+
+		wp_enqueue_script(
+			'wpgraphql-ide',
+			plugins_url( 'build/ide.js', WPGRAPHQL_PLUGIN_FILE ),
+			array_merge( $asset_file['dependencies'], [ 'graphql' ] ),
+			$asset_file['version'],
+			false
+		);
+
+		$localized_data = [
+			'nonce'               => wp_create_nonce( 'wp_rest' ),
+			'graphqlEndpoint'     => trailingslashit( site_url() ) . 'index.php?' . \WPGraphQL\Router::$route,
+			'rootElementId'       => self::WPGRAPHQL_IDE_ROOT_ELEMENT_ID,
+			'context'             => $app_context,
+			'isDedicatedIdePage'  => $this->current_screen_is_dedicated_ide_page(),
+			'dedicatedIdeBaseUrl' => $this->get_dedicated_ide_base_url(),
+		];
+
+		wp_localize_script(
+			'wpgraphql-ide',
+			'WPGRAPHQL_IDE_DATA',
+			$localized_data
+		);
+
+		// Extensions looking to extend GraphiQL can hook in here,
+		// after the window object is established, but before the App renders
+		do_action( 'wpgraphqlide_enqueue_script', $app_context );
+
+		wp_enqueue_script(
+			'wpgraphql-ide-render',
+			plugins_url( 'build/render.js', WPGRAPHQL_PLUGIN_FILE ),
+			array_merge( $asset_file['dependencies'], [ 'wpgraphql-ide', 'graphql' ] ),
+			$render_asset_file['version'],
+			false
+		);
+
+		wp_enqueue_style( 'wpgraphql-ide-app', plugins_url( 'build/ide.css', WPGRAPHQL_PLUGIN_FILE ), [], $asset_file['version'] );
+		wp_enqueue_style( 'wpgraphql-ide-render', plugins_url( 'build/render.css', WPGRAPHQL_PLUGIN_FILE ), [], $asset_file['version'] );
+
+		// Avoid running custom styles through a build process for an improved developer experience.
+		wp_enqueue_style( 'wpgraphql-ide', plugins_url( 'styles/wpgraphql-ide.css', WPGRAPHQL_PLUGIN_FILE ), [], $asset_file['version'] );
+	}
+
+	/**
+	 * Retrieves the base URL for the dedicated WPGraphQL IDE page.
+	 *
+	 * @return string The URL for the dedicated IDE page within the WordPress admin.
+	 */
+	public function get_dedicated_ide_base_url(): string {
+		return menu_page_url( 'graphql-ide', false );
+	}
+
+	/**
+	 * Retrieves the specific header of this plugin.
+	 *
+	 * @param string $key The plugin data key.
+	 * @return string|null The version number of the plugin. Returns an empty string if the version is not found.
+	 */
+	public function get_plugin_header( string $key = '' ): ?string {
+		if ( ! function_exists( 'get_plugin_data' ) ) {
+			require_once ABSPATH . 'wp-admin/includes/plugin.php';
+		}
+
+		if ( empty( $key ) ) {
+			return null;
+		}
+
+		$plugin_data = get_plugin_data( __FILE__ );
+
+		return $plugin_data[ $key ] ?? null;
+	}
```

Comment:
PHPStan: lets enforce the string return type.

```suggestion
		return isset( $plugin_data[ $key ] ) ? (string) $plugin_data[ $key ] : null;
	}
```